pkg/credentialprovider/gcp/metadata.go

```diff
@@ -1,273 +0,0 @@
-// be resolved on GCP.
-func init() {
-	tr := utilnet.SetTransportDefaults(&http.Transport{})
-	metadataHTTPClientTimeout := time.Second * 10
-	httpClient := &http.Client{
-		Transport: tr,
-		Timeout:   metadataHTTPClientTimeout,
-	}
-	credentialprovider.RegisterCredentialProvider("google-dockercfg",
-		&credentialprovider.CachingDockerConfigProvider{
-			Provider: &DockerConfigKeyProvider{
-				MetadataProvider: MetadataProvider{Client: httpClient},
-			},
-			Lifetime: 60 * time.Second,
-		})
-
-	credentialprovider.RegisterCredentialProvider("google-dockercfg-url",
-		&credentialprovider.CachingDockerConfigProvider{
-			Provider: &DockerConfigURLKeyProvider{
-				MetadataProvider: MetadataProvider{Client: httpClient},
-			},
-			Lifetime: 60 * time.Second,
-		})
-
-	credentialprovider.RegisterCredentialProvider("google-container-registry",
-		// Never cache this.  The access token is already
-		// cached by the metadata service.
-		&ContainerRegistryProvider{
-			MetadataProvider: MetadataProvider{Client: httpClient},
-		})
-}
-
-// MetadataProvider is a DockerConfigProvider that reads its configuration from Google
-// Compute Engine metadata.
-type MetadataProvider struct {
-	Client *http.Client
-}
-
-// DockerConfigKeyProvider is a DockerConfigProvider that reads its configuration from a specific
-// Google Compute Engine metadata key: 'google-dockercfg'.
-type DockerConfigKeyProvider struct {
-	MetadataProvider
-}
-
-// DockerConfigURLKeyProvider is a DockerConfigProvider that reads its configuration from a URL read from
-// a specific Google Compute Engine metadata key: 'google-dockercfg-url'.
-type DockerConfigURLKeyProvider struct {
-	MetadataProvider
-}
-
-// ContainerRegistryProvider is a DockerConfigProvider that provides a dockercfg with:
-//
-//	Username: "_token"
-//	Password: "{access token from metadata}"
-type ContainerRegistryProvider struct {
-	MetadataProvider
-}
-
-// Returns true if it finds a local GCE VM.
-// Looks at a product file that is an undocumented API.
-func onGCEVM() bool {
-	var name string
-
-	if runtime.GOOS == "windows" {
-		data, err := exec.Command("wmic", "computersystem", "get", "model").Output()
-		if err != nil {
-			return false
-		}
-		fields := strings.Split(strings.TrimSpace(string(data)), "\r\n")
-		if len(fields) != 2 {
-			klog.V(2).Infof("Received unexpected value retrieving system model: %q", string(data))
-			return false
-		}
-		name = fields[1]
-	} else {
-		data, err := os.ReadFile(gceProductNameFile)
-		if err != nil {
-			klog.V(2).Infof("Error while reading product_name: %v", err)
-			return false
-		}
-		name = strings.TrimSpace(string(data))
-	}
-	return name == "Google" || name == "Google Compute Engine"
-}
-
-// Enabled implements DockerConfigProvider for all of the Google implementations.
-func (g *MetadataProvider) Enabled() bool {
-	onGCE := onGCEVM()
-	if !onGCE {
-		return false
-	}
-	if credentialprovider.AreLegacyCloudCredentialProvidersDisabled() {
```

Comment:
remove the `AreLegacyCloudCredentialProvidersDisabled` function and lock DisableKubeletCloudCredentialProviders to true / GA?

Reply:
Done!